Libraries/LibHTTP/Job.cpp

```diff
@@ -163,9 +206,37 @@ void Job::on_socket_connected()
                     return IterationDecision::Break;
                 }
             }
+
             m_received_buffers.append(payload);
             m_received_size += payload.size();
 
+            if (m_current_chunk_remaining_size.has_value()) {
+                auto size = m_current_chunk_remaining_size.value() - payload.size();
+#ifdef JOB_DEBUG
+                dbg() << "Job: We have " << size << " bytes left over in this chunk";
+#endif
+                if (size == 0) {
+#ifdef JOB_DEBUG
+                    dbg() << "Job: Finished a chunk of " << m_current_chunk_total_size.value() << " bytes";
+#endif
+                    // we've read everything, now let's get the next chunk
+                    size = -1;
+                    auto line = read_line(PAGE_SIZE);
+#ifdef JOB_DEBUG
+                    dbg() << "Line following (should be empty): _" << line << "_";
+#endif
+                    (void)line;
+
+                    if (m_current_chunk_total_size.value() == 0) {
+                        // this was the last chunk, read the rest and discard it.
+                        (void)receive(64 * KB);
```

Comment:
Why is 64 KB the right max number of bytes to receive to "read the rest"?

Reply:
I figured it's big enough for a single chunk.
But this actually leads into another problem I just discovered, chunked encoding allows _more_ headers after the last chunk